transform/utils.go

```diff
@@ -159,16 +159,21 @@ func convertString(raw interface{}) (interface{}, error) {
 	}
 }
 
-func extractTransformInstructions(raw json.RawMessage, transformIdentifier, path string) (*transformInstructions, error) {
+func extractTransformInstructions(raw json.RawMessage, transformIdentifier, path string, instanceType string) (*transformInstructions, error) {
 	rawTransformInstruction, _, _, err := jsonparser.Get(raw, "transform", transformIdentifier)
 	if err != nil && err != jsonparser.KeyPathNotFoundError {
 		return nil, fmt.Errorf("failed to extract raw instance transform: %v", err)
 	} else if len(rawTransformInstruction) == 0 {
 		return nil, nil
 	}
-
 	splits := strings.Split(path, ".")
-	parentPath := strings.Join(splits[:len(splits)-1], ".")
+	var parentPath string
+	// check if last 3 bytes are "[*] and last 4 are not ".[*]"
+	if instanceType == "scalar" && strings.HasSuffix(path, "[*]") && !strings.HasSuffix(path, ".[*]") {
+		parentPath = path
+	} else {
+		parentPath = strings.Join(splits[:len(splits)-1], ".")
```

Comment:
If you're going to move this here then you should move the `splits :=` line too because it isn't used anywhere else but this line.

Reply:
moved.